cmd/forwarder.go

```diff
@@ -76,32 +72,19 @@ func Forwarder() *cobra.Command {
 			fwd := forwarder.NewForwarder(&forwarder.Config{
 				Writer: w,
 			})
-
-			err = reg.Register(fwd)
-			if err != nil {
-				return err
-			}
-
-			d := forwarder.NewDecompressor()
-
-			lis, err := net.Listen("tcp", viper.GetString(flagAddress))
-			if err != nil {
-				return err
-			}
-			// start both gRPC and http servers and return as soon as either of them have an
+			// start both servers and return as soon as either of them have an
 			// error. It is expected that these two servers will either return an error, or run
-			// indefinitely. This will be easier to handle when Prometheus moves away from gRPC
-			// and to an HTTP POST for the generic write API https://github.com/prometheus/prometheus/pull/1957
+			// indefinitely.
 			errCh := make(chan error)
 			once := sync.Once{}
 			go func() {
-				server := grpc.NewServer(grpc.RPCDecompressor(d))
-				remote.RegisterWriteServer(server, fwd)
+				http.Handle("/", forwarder.WriteHandler(fwd, "snappy"))
```

Comment:
Let's start just one http server now that we're accepting push over http instead of using gRPC.